File: src/test/java/org/micoli/php/builders/PathBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.AbstractNode;
import org.micoli.php.tasks.configuration.Path;

public class PathBuilder {
    private final Path path;

    private PathBuilder() {
        this.path = new Path();
    }

    public static PathBuilder create() {
        return new PathBuilder();
    }

    public PathBuilder withLabel(String label) {
        path.label = label;
        return this;
    }

    public PathBuilder withTasks(AbstractNode[] tasks) {
        path.tasks = tasks;
        return this;
    }

    public Path build() {
        return path;
    }
}

File: src/test/kotlin/org/micoli/php/builders/PathBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.AbstractNode
import org.micoli.php.tasks.configuration.Path

class PathBuilder private constructor() {
    private val path: Path = Path()

    fun withLabel(label: String): PathBuilder {
        path.label = label
        return this
    }

    fun withTasks(tasks: Array<AbstractNode>): PathBuilder {
        path.tasks = tasks
        return this
    }

    fun build(): Path = path

    companion object {
        @JvmStatic fun create(): PathBuilder = PathBuilder()
    }
}
